### PR TITLE
Reduce overhead by fixing alignment in SubsetLb

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Reduce overhead by fixing alignment in SubsetLb
Additional Description: Moves many of the smaller members of the class to the end to save small amounts of memory
Risk Level: Low

### CODE REVIEW

File: source/common/upstream/subset_lb.h [hunk 1]
```diff
@@ -426,12 +425,13 @@ class SubsetLoadBalancer : public LoadBalancer, Logger::Loggable<Logger::Id::ups
 
   Stats::Gauge* single_duplicate_stat_{};
 
-  const bool locality_weight_aware_;
-  const bool scale_locality_weight_;
-  const bool list_as_any_;
-
   TimeSource& time_source_;
 
+  const LoadBalancerType lb_type_;
```
Comment (adisuissa) [inline source/common/upstream/subset_lb.h:430]: nit: please add a small note about the coalescing of these data members at the end (something like [this](https://github.com/envoyproxy/envoy/blob/1ac0a5948ca317fd418777dab2ffe27935c0a276/source/common/router/config_impl.h#L342)).

**Review by adisuissa — COMMENTED** [review]
LGTM, modulo small nit.
Thanks!

File: source/common/upstream/subset_lb.h (hunk 1 above)
Comment (DiazAlan) [reply source/common/upstream/subset_lb.h:431]: Done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

/assign-from @envoyproxy/senior-maintainers

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**